src/main/java/ca/corefacility/bioinformatics/irida/model/workflow/submission/AbstractAnalysisSubmission.java

```diff
@@ -0,0 +1,308 @@
+package ca.corefacility.bioinformatics.irida.model.workflow.submission;
+
+import ca.corefacility.bioinformatics.irida.model.IridaResourceSupport;
+import ca.corefacility.bioinformatics.irida.model.MutableIridaThing;
+import ca.corefacility.bioinformatics.irida.model.project.ReferenceFile;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
+import ca.corefacility.bioinformatics.irida.model.user.User;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import org.hibernate.annotations.Type;
+import org.hibernate.envers.Audited;
+import org.hibernate.envers.NotAudited;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.*;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Abstract superclass for analysis submissions.
+ */
+@Entity
+@Table(name = "analysis_submission")
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+@Audited
+@EntityListeners(AuditingEntityListener.class)
+public abstract class AbstractAnalysisSubmission extends IridaResourceSupport implements MutableIridaThing {
+	@Id
+	@GeneratedValue(strategy = GenerationType.AUTO)
+	@Column(name = "id")
+	protected Long id;
+
+	@NotNull
+	@Size(min = 3)
+	@Column(name = "name")
+	protected String name;
+
+	@ManyToOne(fetch = FetchType.EAGER, cascade = CascadeType.DETACH, optional = false)
+	@JoinColumn(name = "submitter", nullable = false)
+	protected User submitter;
```

Comment:
I like having it as the user that created it.  Having the error where a user doesn't have permissions would be very rare as it would be the project owner that creates the automated pipeline.  I can look into how we could disable a pipeline if the permissions are bad, but again I think it'll be very very rare.

Reply:
I think it will still come up (e.g., multiple people can be given `MANAGER` permission on a project, but eventually some may leave a project, which could include someone who originally set up the automated pipelines).

I still do think it's a good idea for the pipeline to be submitted as the user who created it though.